hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/index/bloom/TestHoodieGlobalBloomIndex.java

```diff
@@ -297,6 +297,8 @@ public void testTagLocation() throws Exception {
 
     for (HoodieRecord record : taggedRecordRDD.collect()) {
       IndexedRecord data = record.toIndexedRecord(SIMPLE_RECORD_SCHEMA, CollectionUtils.emptyProps()).get().getData();
+      // eager deser.
+      data.get(0);
```

Comment:
Why do we need this?

Reply:
Could we update the `equals` method to deserialize the record? Would that make the testing code cleaner?